addons/product/models/product.py

```diff
@@ -330,7 +330,9 @@ def unlink(self):
             if not product.exists():
                 continue
             # Check if the product is last product of this template
-            other_products = self.search([('product_tmpl_id', '=', product.product_tmpl_id.id), ('id', '!=', product.id)])
+            other_products = self.with_context(active_test=False).search([
+                ('product_tmpl_id', '=', product.product_tmpl_id.id), 
+                ('id', 'not in', self.ids)])
```

Comment:
as we are in a for product in self loop:
no need to change the second condition of the search domain ;).